tests/base/src/test/java/org/wildfly/security/ssl/TLS13AuthenticationTest.java

```diff
@@ -114,12 +116,69 @@ public void testTwoWayTLS13() throws Exception {
         assertNotNull(identity);
         assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
     }
 
-    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, String clientConfigFileName, String expectedCipherSuite) throws Exception {
+    @Test
+    public void testDifferentPreferredTLS13Suites() throws Exception {
+        final String REQUIRED_CIPHER_SUITE = "TLS_AES_128_GCM_SHA256";
+        final String PREFERRED_CIPHER_SUITE = "TLS_AES_256_GCM_SHA384";
+
+        SSLContext serverContext = new SSLContextBuilder()
+                .setSecurityDomain(securityDomain)
+                .setCipherSuiteSelector(CipherSuiteSelector.fromNamesString(REQUIRED_CIPHER_SUITE).add(PREFERRED_CIPHER_SUITE))
+                .setKeyManager(getKeyManager("/jks/scarab.keystore"))
+                .setTrustManager(getCATrustManager())
+                .setNeedClientAuth(true)
+                .build().create();
+
+        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-different-preferred-tls13-suites.org", "wildfly-ssl-test-config-v1_5.xml", REQUIRED_CIPHER_SUITE, true);
+        assertNotNull(identity);
+        assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
+    }
+
+    @Test
+    public void testServerOnlyTLS13() throws Exception {
+        final String SERVER_CIPHER_SUITE = "TLS_AES_128_GCM_SHA256";
+        final String CLIENT_CIPHER_SUITE = "TLS_RSA_WITH_AES_128_CBC_SHA256"; // TLS v1.2
+
+        SSLContext serverContext = new SSLContextBuilder()
+                .setSecurityDomain(securityDomain)
+                .setCipherSuiteSelector(CipherSuiteSelector.aggregate(
+                                CipherSuiteSelector.fromNamesString(SERVER_CIPHER_SUITE),
+                                CipherSuiteSelector.fromString(CLIENT_CIPHER_SUITE)
+                ))
+                .setKeyManager(getKeyManager("/jks/scarab.keystore"))
+                .setTrustManager(getCATrustManager())
+                .setNeedClientAuth(true)
+                .build().create();
+
+        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-server-only-tls13.org", "wildfly-ssl-test-config-v1_5.xml", CLIENT_CIPHER_SUITE, false);
+        assertNotNull(identity);
+        assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
+    }
+
+    @Test
+    public void testClientOnlyTLS13() throws Exception {
+        final String SERVER_CIPHER_SUITE = "TLS_RSA_WITH_AES_128_CBC_SHA256"; // TLS v1.2
+
+        SSLContext serverContext = new SSLContextBuilder()
+                .setSecurityDomain(securityDomain)
+                .setCipherSuiteSelector(CipherSuiteSelector.fromString(SERVER_CIPHER_SUITE))
+                .setProtocolSelector(ProtocolSelector.DEFAULT_SELECTOR.remove("TLSv1.3"))
```

Comment:
I think this line can be removed. Whether or not TLSv1.3 is enabled depends on the cipher suites that are configured. So if no TLSv1.3 cipher suites have been specified, then TLSv1.3 won't be enabled.